src/link/index.ts

```diff
@@ -266,14 +268,22 @@ async function linkAllPkgs (
   alldeps: DependencyTreeNode[],
   opts: {
     force: boolean,
+    sideEffectsCache: boolean,
   }
 ) {
   return Promise.all(
     alldeps.map(async pkg => {
       const filesResponse = await pkg.fetchingFiles
 
       if (pkg.independent) return
-      return storeController.importPackage(pkg.centralLocation, pkg.peripheralLocation, {
+
+      let importFrom = pkg.centralLocation
```

Comment:
just change the value of `centralLocation` to the path that will be used here, so move this logic with sideEffectsCache to `resolvePeers` and assign `centralLocation` the correct value